yb-voyager/cmd/analyzeSchema.go

```diff
@@ -378,1090 +334,21 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
-	*/
-	if len(createIndexNode.IndexStmt.GetIndexParams()) > 1 {
-		summaryMap["INDEX"].invalidCount[displayObjectName] = true
-		reportCase(fpath, "Schema contains gin index on multi column which is not supported.",
-			"https://github.com/yugabyte/yugabyte-db/issues/10652", "", "INDEX", displayObjectName,
-			sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, GIN_INDEX_MULTI_COLUMN_DOC_LINK)
-		return
-	}
-	/*
-		e.g. CREATE INDEX idx_name ON public.test USING gin (data DESC);
-		stmt:{index_stmt:{idxname:"idx_name" relation:{schemaname:"public" relname:"test" inh:true relpersistence:"p" location:44}
-		access_method:"gin" index_params:{index_elem:{name:"data" ordering:SORTBY_DESC nulls_ordering:SORTBY_NULLS_DEFAULT}}}} stmt_len:80
-	*/
-	idxParam := createIndexNode.IndexStmt.GetIndexParams()[0] // taking only the first as already checking len > 1 above so should be fine
-	if idxParam.GetIndexElem().GetOrdering() != pg_query.SortByDir_SORTBY_DEFAULT {
-		summaryMap["INDEX"].invalidCount[displayObjectName] = true
-		reportCase(fpath, "Schema contains gin index on column with ASC/DESC/HASH Clause which is not supported.",
-			"https://github.com/yugabyte/yugabyte-db/issues/10653", "", "INDEX", displayObjectName,
-			sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, GIN_INDEX_DIFFERENT_ISSUE_DOC_LINK)
-	}
-
-}
-
-func reportPolicyRequireRolesOrGrants(createPolicyNode *pg_query.Node_CreatePolicyStmt, sqlStmtInfo sqlInfo, fpath string) {
-	policyName := createPolicyNode.CreatePolicyStmt.GetPolicyName()
-	roles := createPolicyNode.CreatePolicyStmt.GetRoles()
-	relname := createPolicyNode.CreatePolicyStmt.GetTable()
-	schemaName := relname.Schemaname
-	tableName := relname.Relname
-	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
-	roleNames := make([]string, 0)
-	/*
-		e.g. CREATE POLICY P ON tbl1 TO regress_rls_eve, regress_rls_frank USING (true);
-		stmt:{create_policy_stmt:{policy_name:"p" table:{relname:"tbl1" inh:true relpersistence:"p" location:20} cmd_name:"all"
-		permissive:true roles:{role_spec:{roletype:ROLESPEC_CSTRING rolename:"regress_rls_eve" location:28}} roles:{role_spec:
-		{roletype:ROLESPEC_CSTRING rolename:"regress_rls_frank" location:45}} qual:{a_const:{boolval:{boolval:true} location:70}}}}
-		stmt_len:75
-
-		here role_spec of each roles is managing the roles related information in a POLICY DDL if any, so we can just check if there is
-		a role name available in it which means there is a role associated with this DDL. Hence report it.
-
-	*/
-	for _, role := range roles {
-		roleName := role.GetRoleSpec().GetRolename() // only in case there is role associated with a policy it will error out in schema migration
-		if roleName != "" {
-			//this means there is some role or grants used in this Policy, so detecting it
-			roleNames = append(roleNames, roleName)
+		err = parserIssueDetector.ParseRequiredDDLs(sqlStmtInfo.formattedStmt)
```

Comment:
What you can do here is:
1. Store the parse tree from line 328
2. Pass the parse tree directly to ParseRequiredDDLs()
3. Internally, Change ParseAndProcessDDL() -> ProcessDDL()
4. Also rename ParseRequiredDDLs() -> ProcessDDLs() or ProcessRequiredDDLs()

Reply:
I would not like to introduce the `pg_query` library in the `queryissue` pkg because we already have the `queryparser`, so will call the Parse in the `ParseRequiredDDLs`.